pkg/operator/ceph/disruption/machinedisruption/reconcile.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package machinedisruption
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/coreos/pkg/capnslog"
+	healthchecking "github.com/openshift/machine-api-operator/pkg/apis/healthchecking/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	cephClient "github.com/rook/rook/pkg/daemon/ceph/client"
+	cephCluster "github.com/rook/rook/pkg/operator/ceph/cluster"
+	"github.com/rook/rook/pkg/operator/ceph/disruption/controllerconfig"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	controllerName                  = "machinedisruption-controller"
+	MDBNamespace                    = "openshift-machine-api"
+	MDBCephClusterNamespaceLabelKey = "rook.io/cephClusterNamespace"
+	MDBCephClusterNameLabelKey      = "rook.io/cephClusterName"
+	MDBFencingLabelKey              = "fencegroup.rook.io/cluster"
+)
+
+var logger = capnslog.NewPackageLogger("github.com/rook/rook", controllerName)
+
+// ReconcileMachineDisruption reconciles MachineDisruption
+type ReconcileMachineDisruption struct {
+	scheme  *runtime.Scheme
+	client  client.Client
+	log     *capnslog.PackageLogger
+	context *controllerconfig.Context
+}
+
+const (
+	defaultMaxUnavailable   = 1
```

Comment:
Set this to 0 so that we can't fence until we know the ceph health.

Reply:
done